services/user-tenant-service/src/test/java/com/sourcefuse/jarc/services/usertenantservice/integration/GroupControllerTests.java

```diff
@@ -0,0 +1,504 @@
+
+    // Verify that roleRepository.count() was called
+    Mockito.verify(groupRepository).count();
+  }
+
+  @Test
+  @DisplayName("Test: countGroup with database error")
+   void testCountGroup_DatabaseError() throws Exception {
+    // Mock the behavior of roleRepository.save() to throw an exception
+    Mockito
+      .when(groupRepository.count())
+      .thenThrow(new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR));
+
+    // Perform the API call and catch the exception
+    mockMvc
+      .perform(
+        MockMvcRequestBuilders
+          .get(basePath + "/count")
+          .contentType(MediaType.APPLICATION_JSON)
+      )
+      .andExpect(MockMvcResultMatchers.status().isInternalServerError())
+      .andExpect(result ->
+        Assertions.assertTrue(
+          result.getResolvedException() instanceof ResponseStatusException
+        )
+      )
+      .andExpect(result ->
+        Assertions.assertEquals(
+          HttpStatus.INTERNAL_SERVER_ERROR.value(),
+          result.getResponse().getStatus()
+        )
+      );
+
+    // Verify that roleRepository.save() was called
+    Mockito.verify(groupRepository).count();
+  }
+
+  @Test
+  @DisplayName("Test getAllGroups success")
+   void testGetAllGroups_Success() throws Exception {
+    // Prepare test data
+    List<Group> groups = Arrays.asList(
+      new Group(),
+      new Group(),
+      new Group()
+    );
+
+    // Mock the behavior of roleRepository.findAll()
+    Mockito.when(groupRepository.findAll()).thenReturn(groups);
+
+    // Perform the API call
+    mockMvc
+      .perform(
+        MockMvcRequestBuilders
+          .get(basePath)
+          .contentType(MediaType.APPLICATION_JSON)
+      )
+      .andExpect(MockMvcResultMatchers.status().isOk())
+      .andExpect(result -> Assertions.assertNotNull(result.getResponse()))
+      .andExpect(MockMvcResultMatchers.jsonPath("$.size()").value(3));
+    // Verify that roleRepository.findAll() was called
+    Mockito.verify(groupRepository).findAll();
+  }
+
+  @Test
+  @DisplayName("Test: getAllGroups Empty Response")
+   void testGetAllGroup_Empty() throws Exception {
+    // Mock the behavior of roleRepository.findAll()
+    Mockito.when(groupRepository.findAll()).thenReturn(Arrays.asList());
+
+    // Perform the API call
+    mockMvc
+      .perform(
+        MockMvcRequestBuilders
+          .get(basePath)
+          .contentType(MediaType.APPLICATION_JSON)
+      )
+      .andExpect(MockMvcResultMatchers.status().isOk())
+      .andExpect(result -> Assertions.assertNotNull(result.getResponse()))
+      .andExpect(MockMvcResultMatchers.jsonPath("$.size()").value(0));
+
+    // Verify that roleRepository.findAll() was called
+    Mockito.verify(groupRepository).findAll();
+  }
+
+  @Test
+  @DisplayName("Test: GetGroupByID with existing group")
+   void testGetGroupByID_ExistingRole() throws Exception {
+    //get Current User
+    MockCurrentUserSession.setCurrentLoggedInUser(null, null, null);
```

Comment:
Move this code in beforeEach don't need to repeat this as all API's should be authenticated